src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -178,6 +181,27 @@ export class SavedObjectsRepository {
     const method = id && !overwrite ? 'create' : 'index';
     const time = this._getCurrentTime();
 
+    if (predicate != null) {
```

Comment:
I have to give further thought into how these "predicates" should work with regard to migrations. It seems like we'd want to ensure that the migrations are applied to the "new saved-object" and the "existing saved-object" prior to running them through the predicate, but I struggled with all of the various types we use internally...